common/authorization/authorizer.go

```diff
@@ -35,6 +35,15 @@ const (
 	DecisionAllow
 )
 
+const (
+	// PermissionRead means the user can write on the domain level APIs
+	PermissionRead Permission = iota + 1
```

Comment:
For internal enums it makes sense to use iota

But I think we should keep these as strings. Because this is an external facing constant. Other library like webUI/SDKs of Java/Golang would be hard to follow this integer.

Reply:
ahh nvm, I saw your `NewPermission`. LGTM

Reply:
Had my questions when I changed it from string to iota. The reason I did this was to evaluate the permissions based on the integer instead of doing

```
if admin:
   blabla
if write and permission in [write, read]:
 blabla
elif read and permission == read:
  blabla
else:
  deny
 ```
  This is not really scalable, however do you think more permissions might be added in a future? I'm still thinking if integer is the right call here

Reply:
I like the approach of integer for comparison. For now I don’t see any need of more permission.